### PR TITLE
ARTEMIS-3652 - fix test cleanup, allow embedded broker properties to …

### PR DESCRIPTION
**Pull request by gtully**
…be configured, follows up on ARTEMIS-3627

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java [hunk 1]
```diff
@@ -144,9 +155,11 @@ protected void initStart() throws Exception {
...
       }
 
-      URL brokerPropertiesFromClasspath = this.getClass().getClassLoader().getResource(ActiveMQDefaultConfiguration.BROKER_PROPERTIES_SYSTEM_PROPERTY_NAME);
-      if (brokerPropertiesFromClasspath != null) {
-         activeMQServer.setProperties(new File(brokerPropertiesFromClasspath.toURI()).getAbsolutePath());
+      if (propertiesResourcePath != null) {
+         URL brokerPropertiesFromClasspath = this.getClass().getClassLoader().getResource(propertiesResourcePath);
+         if (brokerPropertiesFromClasspath != null) {
+            activeMQServer.setProperties(new File(brokerPropertiesFromClasspath.toURI()).getAbsolutePath());
+         }
```
Comment (gemmellr) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java:162]: Feels like this should really be in a separate commit against ARTEMIS-3627 specifically, where the original logic was only just added.

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java [hunk 2]
```diff
@@ -398,7 +407,7 @@ private void prepareNodesAndStartCombinedHeadTail() throws Exception {
          configuration.setName("Node" + nodeId);
          configuration.setBrokerInstance(new File(getTestDirfile(), configuration.getName()));
          configuration.addAcceptorConfiguration("tcp", "tcp://localhost:" + (base_port + (nodeId)) + "?redirect-to=" + balancerConfigName + ";amqpCredits=1000;amqpMinCredits=300");
-         nodes.add(new EmbeddedActiveMQ().setConfiguration(configuration));
+         nodes.add(new EmbeddedActiveMQ().setConfiguration(configuration).setPropertiesResourcePath(null));
```
Comment (gemmellr) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java:410]: This seems odd...is it actually necessary? Is there a default file on the classpath screwing with this test in some way? That would seem like a general issue needing fixed in itself (also against ARTEMIS-3627 perhaps?), it really doesnt feel like unrelated individual tests should need to be randomly setting this null to work.

Comment (gtully) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java:410]: It does need a way to turn it off so setting null is good. 
Yes there is a file on the classpath, that changed the jmx name, I will fix that in another commit

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java (hunk 1 above)
Comment (gtully) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java:162]: It will get referenced via the commit message

Comment (gemmellr) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java:162]: It should be the primary target of the commit, and shouldnt have unrelated test changes for another issue.

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java (hunk 2 above)
Comment (gemmellr) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java:410]: Having a way to do it is fine, but it doesnt seem like this test (or mostly any other) should need to do so.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java (hunk 1 above)
Comment (gtully) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/embedded/EmbeddedActiveMQ.java:162]: pushed https://github.com/apache/activemq-artemis/pull/3924 with this logic and sorts the test failure. thanks!

**Comment by gtully** [comment]
seeing intermittent failure of elastic queue test now, need to investigate what is going in. the name property introduced by ARTEMIS-3652  were the cause of failure, so there must be something else in play.

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java (hunk 2 above)
Comment (gtully) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/balancing/ElasticQueueTest.java:410]: agree. thanks.

**Comment by gtully** [comment]
tracking connection errors via a listener and in onMessage and dealing with a missed send reply for correlation of received messages. cannot reproduce any failure now.

**Comment by gemmellr** [comment]
Seems reasonable.
